### DISCUSSION TITLE
Not user friendly error for running ad-hoc command with no setup

### DISCUSSION THREAD

**Discussion by bessarabov** [umputun/spot#39] [General]
This is a very minor thing.

I do not specify inventory, there is no playbook and no --target, but I try to run ad-hoc command.

I get the error (this is expected, because the system does not know where to execute the command), but the message can be more user friendly (and I"m not sure that usage of `default` hostname makes much sense)

```
bessarabov@bessarabov-osx:~$ env -i ~/bin/spot ls
spot v0.9.1-0aa555f-2023-05-02T08:06:47Z
[default:22]  ! can't connect to default:22: failed to create client connection to default:22: failed to dial: dial tcp: lookup default: no such host
failed, 1 error occurred:
        * can't run task "ad-hoc" for target "default": 1 error(s) occurred: [0] {can't connect to default:22: failed to create client connection to default:22: failed to dial: dial tcp: lookup default: no such host}

bessarabov@bessarabov-osx:~$
```

**Comment by umputun** [comment]
Yeah, this is unclear, but I'm unsure what to do here. This is a "failed match" that found nothing suitable in the inventory/playbook and used the "default" target name. Removing "default" logic for ad-hoc command is not a good idea either, as running ad-hoc with inventory is a valid use case

**Comment by bessarabov** [comment]
Thank you for the explanation. I was thinking about this and I can't propose anything what will be better that the current behaviour. (at first I was thinking to threat this as a some special case and output a better message, but now I think that this message can give an interesting idea to the person who is reading it — add `SOME_IP default` to /etc/hosts to make it possible to work via spot with the host where you work most of the time)

So, I'm closing this discussion.
